posthog/api/session_recording.py

```diff
@@ -136,19 +138,51 @@ def delete(self, request: request.Request, *args: Any, **kwargs: Any) -> Respons
 
         return Response({"success": True})
 
+    @action(methods=["GET"], detail=True)
+    def snapshot_file(self, request: request.Request, **kwargs) -> HttpResponse:
+        file_key = request.GET.get("blob_key")
+
+        if not file_key:
+            raise exceptions.ValidationError("Must provide a snapshot file blob key")
+
+        url = object_storage.get_presigned_url(file_key)
+        if not url:
+            raise exceptions.NotFound("Snapshot file not found")
+
+        with requests.get(url=url, stream=True) as r:
+            r.raw.decode_content = True
+            r.raise_for_status()
+            response = HttpResponse(content=r.raw, content_type="application/json")
+            response["Content-Disposition"] = "inline"
+            return response
+
     # Paginated endpoint that returns the snapshots for the recording
     @action(methods=["GET"], detail=True)
     def snapshots(self, request: request.Request, **kwargs):
-        # TODO: Why do we use a Filter? Just swap to norma, offset, limit pagination
-        filter = Filter(request=request)
-        limit = filter.limit if filter.limit else DEFAULT_RECORDING_CHUNK_LIMIT
-        offset = filter.offset if filter.offset else 0
-
         recording = SessionRecording.get_or_build(session_id=kwargs["pk"], team=self.team)
 
         if recording.deleted:
             raise exceptions.NotFound("Recording not found")
 
+        if request.GET.get("blob_loading_enabled", "false") == "true":
+            blob_keys = object_storage.list_objects(
```

Comment:
Related to above - this should only return the final parts of the file, not the entire file path